### ISSUE TITLE
Bug: File modification indicator ([+]) not updated for inactive windows

### ISSUE THREAD

**Issue by jaekyeom** [hoob3rt/lualine.nvim#157]
Thanks for the great project 👍

Btw, it looks that the indicator for file modification, [+], doesn't get updated for inactive windows.
The following gifs show some demonstrations.

![filemodified](https://user-images.githubusercontent.com/6262727/113230768-b1e18100-92d4-11eb-8008-3c343c1e5ea4.gif)

![diffedit](https://user-images.githubusercontent.com/6262727/113230982-2ae0d880-92d5-11eb-95bb-259c20dae24f.gif)

**Comment by shadmansaleh** [comment]
Well actually non of the component is updated in inactive windows . We freeze the statusline on WinLeave . It was done to resolve #32 . We are open to better soloution but updating inactive statusline seems to be a hassle as components needs to hold their state and there's single component for all buffers and window ;)

**Comment by hoob3rt** [comment]
Just as @shadmansaleh said, the statusline does not update on inactive windows. And for me that's acceptable. I don't see why there would be a need to update the statusline for a file in which you are not focusing on but I'm open to discuss this further.

**Comment by jaekyeom** [comment]
Thanks for opening the discussion.

Probably one thing is consistency.
Since this plugin provides the "status"line, it's expected to display the status of vim files, buffers and windows.
In the gifs I've attached, giving the focus to the left windows didn't change the status of them, but the statusline updated.

But practically, a bigger issue to me is when I'm putting diff chunks to the opposite window using `dp` or `:diffput` (the second gif).
Sometimes I want to check if I put changes and/or saved the file by just looking at the statusline ;)

**Comment by shadmansaleh** [comment]
> But practically, a bigger issue to me is when I'm putting diff chunks to the opposite window using dp or :diffput (the second gif).
Sometimes I want to check if I put changes and/or saved the file by just looking at the statusline ;)

Seems resonable . I might take a look after #141 is merged . Though no promises :D

**Comment by jaekyeom** [comment]
I appreciate the consideration and your efforts! :)

**Comment by shadmansaleh** [comment]
I've tried it . Like I said before It seems too much of a hassle . The ideas I had were storeing component results in buf local variables insted of srcipt local ones currently used / creating new instences of components for each buffer and make component instences not have any static variables (It has issues as for some components autocomands are used to update them ) . These add a ton of overhead for each iteration of statusline . And in my opinion updating statusline for inactive buffer has little to gain .
If any one has better ideas to do this you can suggest .

@jaekyeom Did statusline you used before lualine update on inactive buffers . If it did can you say what statusline you used . I might checkout how they were doing it :)

**Comment by jaekyeom** [comment]
@shadmansaleh Thanks for looking into it!
Actually, since I've recently started switching to lua-based plugins, I have no much experience with other lua-based statusline plugins.
Before lualine, I was using [vim-airline](https://github.com/vim-airline/vim-airline), and it does support update on inactive windows (not sure about *buffers*, though).

**Comment by hoob3rt** [comment]
One idea to do this is have a flag which components should update in inactive buffers. But I think it will complicate the codebase by a lot and most likely create more bugs. airline obviously does this because they handle every edge case (their codebase is obviously more mature than this one). I won't close this because eventually this should be resolved but there is a lot of other stuff I want to do with this project before that so I cannot give you an ETA on when this could be done but I surely won't touch this before we have a proper test suite for the whole project.

**Comment by shadmansaleh** [comment]
> One idea to do this is have a flag which components should update in inactive buffers.

That's not the issue . It's really easy to add a condition on draw function. And we can just pass the is_focused from statusline(init.lua) -> draw_colomponent(utils.lua) -> draw(component.lua) . That doesn't add much complication . But the real problem it shared stat of components for all buffers. And separating them adds complication .

**Comment by hoob3rt** [comment]
Yeah I wasn't clear enough. To do this we must separate statusline on windows (hard part) and should update only a few components (I think). But as I've said the priority for me is cleaning up other pr's/issues and getting test suite to a viable state.

**Comment by shadmansaleh** [comment]
Fixed in #311
